fastly/data_source_fastly_tls_activation.go

```diff
@@ -0,0 +1,167 @@
+package fastly
+
+import (
+	"fmt"
+	"github.com/fastly/go-fastly/v3/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"time"
+)
+
+func dataSourceFastlyTLSActivation() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceFastlyTLSActivationRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				Description:   "Fastly Activation ID. Conflicts with all other filters.",
+				ConflictsWith: []string{"certificate_id", "configuration_id", "domain"},
```

Comment:
We used this pattern in all the singular data sources, I believe, to enforce that _either_ the `id` field is used, _or_ 0-3 of the other filters, but not both. This is so that the `id` field can be used in conjunction with the corresponding plural data source, in which case it wouldn't make any sense to specify another filter as it's just a simple ID lookup. Similarly, it wouldn't make sense for a user to specify an ID if they were trying to filter based on another attribute.

So you're right in that it's just to get a nice Terraform error for the user, but it's less about enforcing they're all set, just that none of those listed are also set when `id` is present. Put differently, `ConflictsWith` doesn't care how many of the listed fields are set, as long as none of them are set when `id` is.